builtin/k8s/platform.go

```diff
@@ -296,318 +294,392 @@ func (p *Platform) resourceDeploymentStatus(
+			q, err := k8sresource.ParseQuantity(cpu.Requested)
 			if err != nil {
-				return err
+				return nil,
+					status.Errorf(codes.InvalidArgument, "failed to parse cpu request %s to k8s quantity: %s", cpu.Requested, err)
 			}
-
 			resourceRequests[corev1.ResourceCPU] = q
 		}
 
-		if p.config.CPU.Limit != "" {
-			q, err := k8sresource.ParseQuantity(p.config.CPU.Limit)
+		if cpu.Limit != "" {
+			q, err := k8sresource.ParseQuantity(cpu.Limit)
 			if err != nil {
-				return err
+				return nil,
+					status.Errorf(codes.InvalidArgument, "failed to parse cpu limit %s to k8s quantity: %s", cpu.Limit, err)
 			}
-
 			resourceLimits[corev1.ResourceCPU] = q
 		}
 	}
 
-	if p.config.Memory != nil {
-		if p.config.Memory.Requested != "" {
-			q, err := k8sresource.ParseQuantity(p.config.Memory.Requested)
+	if memory != nil {
+		if memory.Requested != "" {
+			q, err := k8sresource.ParseQuantity(memory.Requested)
 			if err != nil {
-				return err
+				return nil,
+					status.Errorf(codes.InvalidArgument, "failed to parse memory requested %s to k8s quantity: %s", memory.Requested, err)
 			}
-
 			resourceRequests[corev1.ResourceMemory] = q
 		}
 
-		if p.config.Memory.Limit != "" {
-			q, err := k8sresource.ParseQuantity(p.config.Memory.Limit)
+		if memory.Limit != "" {
+			q, err := k8sresource.ParseQuantity(memory.Limit)
 			if err != nil {
-				return err
+				return nil,
+					status.Errorf(codes.InvalidArgument, "failed to parse memory limit %s to k8s quantity: %s", memory.Limit, err)
 			}
-
 			resourceLimits[corev1.ResourceMemory] = q
 		}
 	}
 
-	for k, v := range p.config.Resources {
-		if strings.HasPrefix(k, "limits_") {
-			limitKey := strings.Split(k, "_")
-			resourceName := corev1.ResourceName(limitKey[1])
+	for k, v := range resources {
+		if strings.HasPrefix(k, "limits_") || strings.HasPrefix(k, "requests_") {
+			key := strings.Split(k, "_")
+			resourceName := corev1.ResourceName(key[1])
 
 			quantity, err := k8sresource.ParseQuantity(v)
 			if err != nil {
-				return err
+				return nil,
+					status.Errorf(codes.InvalidArgument, "failed to parse resource %s to k8s quantity: %s", v, err)
 			}
 			resourceLimits[resourceName] = quantity
-		} else if strings.HasPrefix(k, "requests_") {
-			reqKey := strings.Split(k, "_")
-			resourceName := corev1.ResourceName(reqKey[1])
-
-			quantity, err := k8sresource.ParseQuantity(v)
-			if err != nil {
-				return err
-			}
-			resourceRequests[resourceName] = quantity
 		} else {
 			log.Warn("ignoring unrecognized k8s resources key: %q", k)
 		}
 	}
 
-	_, cpuLimit := resourceLimits[corev1.ResourceCPU]
-	_, cpuRequest := resourceRequests[corev1.ResourceCPU]
-
-	if p.config.AutoscaleConfig != nil && !(cpuLimit || cpuRequest) {
-		ui.Output("For autoscaling in Kubernetes to work, a deployment must specify "+
-			"cpu resource limits and requests. Otherwise the metrics-server will not properly be able "+
-			"to scale your deployment.", terminal.WithWarningStyle())
-	}
-
```

Comment:
This shouldn't of been deleted. Notice we still need to check the cpu limit and requests check.